search/src/main/java/com/connexta/search/query/controllers/QueryController.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.search.query.controllers;
+
+import com.connexta.search.query.QueryManager;
+import com.connexta.search.query.exceptions.QueryException;
+import java.net.URI;
+import java.util.List;
+import javax.validation.ConstraintViolationException;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@RestController
+@RequestMapping(value = "/search")
+@AllArgsConstructor
+@Validated
+@Slf4j
+public class QueryController {
+
+  private QueryManager queryManager;
+
+  @GetMapping
+  @ResponseBody
+  public ResponseEntity<List<URI>> searchKeyword(
+      @RequestParam(value = "q") @NotBlank final String keyword) {
+    try {
+      return new ResponseEntity<>(queryManager.find(keyword), HttpStatus.OK);
+    } catch (QueryException e) {
+      log.warn("Unable to search for {}", keyword, e);
+    }
+
+    return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+  }
+
+  @ControllerAdvice
+  private class ConstraintViolationExceptionHandler extends ResponseEntityExceptionHandler {
```

Comment:
If the @ExceptionHandler is in defined in the controller class, the private class and @ControllerAdvice are unnecessary.

Reply:
I agree, but I just copied this from the mis repo. We can fix it in a future PR.

Reply:
`handleExceptionInternal` is defined in ResponseEntityExceptionHandler, so I can't not extend that class, right? In any case, I just copied this from the mis repo, so it can be improved in a future PR to follow what you implemented in https://github.com/connexta/ion-store/pull/148.